wren/service/materialization.go

```diff
@@ -266,12 +307,20 @@ func (s *materializationService) DeleteMaterialization(ctx context.Context, requ
 }
 
 func (s *materializationService) deleteMaterialization(ctx context.Context, owner *ent.Owner, request *v1alpha.DeleteMaterializationRequest) (*v1alpha.DeleteMaterializationResponse, error) {
-	materialization, err := s.materializationClient.GetMaterializationByName(ctx, owner, request.MaterializationName)
+	foundMaterialization, err := s.materializationClient.GetMaterializationByName(ctx, owner, request.MaterializationName)
 	if err != nil {
 		return nil, err
 	}
 
-	err = s.materializationClient.DeleteMaterialization(ctx, owner, materialization)
+	if foundMaterialization.SourceType == materialization.SourceTypeStreams {
```

Comment:
I assume in the future we'll also have a way to have a long-running file backed materialization, so maybe a `todo` here